### DISCUSSION TITLE
PC-BASIC release 2.0.7

### DISCUSSION THREAD

**Discussion by robhagemans** [robhagemans/pcbasic#224] [Announcements]
For desktop use, use one of the installers here. 
For Python use, install with  `pip install pcbasic`.

With this release, **the 2.0 branch of PC-BASIC is now in maintenance mode**.   
New development will take place on a new 2.1 branch, which will drop support for deprecated features and old Python and OS versions.

Please note:
- The Mac package is only _ad-hoc codesigned_. If you have problems with the install, please ensure you have convinced your Mac's Gatekeeper to allow installation of such software.
- When installing with `apt` on Ubuntu, please ensure you have enabled the "universe" repository where the dependencies of this package reside.
- The `.rpm` package has been discontinued, I've not been able to make it work.

## Selected improvements
- Fixes issues #218, #219, #212, #208
- Fixes an issue with screen updates on Windows (issue #166, mark 2)
- Fixes various compatibility issues with `WINDOW`ed graphics
- Fixes issues with redirection of printer output and standard input
- Enables 12 function keys through `KEY` 30, 31 in `syntax=advanced` (BASICA), use `syntax=gwbasic` to disable
- Improves test coverage
- Improves crash reporting

## SHA-256 hashes

    4c28e92654a405e61152eaba3bd0317534dcbb51ef8d4966f2e3caf844abb261  PC-BASIC-2.0.7.dmg
    e295374656965f8d24ca34765bbce4b7762133f4ef94ac9bb58b3396f4b9b46c  PC-BASIC-2.0.7.msi
    586fd172fbe08cb1f4a05c980177abba3e983229c538b9e531f958719a4f90e7  pcbasic-2.0.7-py2.py3-none-any.whl
    34d0236064ea7a4a2962cacd5e668812cf360b3d8aa18009f94dfb61027355ec  pcbasic-2.0.7.tar.gz
    985e9e1193e835936ccbbf847980839975f601ccc60b5755a8c075015067c044  python3-pcbasic_2.0.7_all.deb

<hr /><em>This discussion was created from the release <a href='https://github.com/robhagemans/pcbasic/releases/tag/'>PC-BASIC release 2.0.7</a>.</em>

**Comment by toml12953** [comment]
I've installed PC-BASIC 2.07 in Windows but how do I run it? Typing pcbasic at a command prompt can't find the file and does nothing.

**Comment by robhagemans** [comment]
If you I stalled the Windows package, it'll be in your start menu so you
can just click it.

On the command prompt you may need to type the full location which may
depend on your local setup - try looking in C:\Program Files (x86)\PC-BASIC
2.0 or similarly named locations.

**Comment by gdfaini** [comment]
Thank you so much!  After 30+ years I can open, edit, and run my Z-BASIC Star Trek game.  The emulator works fine, but I get the attached error every time I start it, either from Start menu or desktop shortcut. After I press Continue it works properly under Windows 7.
![No Disk Error](https://github.com/user-attachments/assets/e293917b-715b-4456-9504-1e5b5fb76b92)

**Reply by Marrin** [reply]
For reporting errors it is usually better to open an issue. 🙂

**Reply by gdfaini** [reply]
Thanks.

**Comment by dfileccia** [comment]
I was comparing PC-BASIC to GW-BASIC and it is astounding how accurate it is. Even down to the free memory!
